File: java/parser/src/main/java/gov/lbl/parser/parser/Constraining_clauseVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Class_modification;
import gov.lbl.parser.domain.Constraining_clause;
import gov.lbl.parser.domain.Name;

public class Constraining_clauseVisitor extends modelicaBaseVisitor<Constraining_clause> {
    @Override
    public Constraining_clause visitConstraining_clause(modelicaParser.Constraining_clauseContext ctx) {
        NameVisitor nameVisitor = new NameVisitor();
        Name name = ctx.name() == null ? null : ctx.name().accept(nameVisitor);
        
        Class_modificationVisitor class_modificationVisitor = new Class_modificationVisitor();
        Class_modification class_modification = ctx.class_modification() == null ? null : ctx.class_modification().accept(class_modificationVisitor);

        return new Constraining_clause(name, class_modification);
    }
}

File: jsParser/parser/Constraining_clauseVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor;
const Constraining_clause = require('../domain/Constraining_clause');

const NameVisitor = require('./NameVisitor');
const Class_modificationVisitor = require('./Class_modificationVisitor');

class Constraining_clauseVisitor {
    constructor() {
        modelicaVisitor.call(this);
        return this;
    }
    visitConstraining_clause(ctx) {
        var name = null;
        var class_modification = null;
        
        if (ctx.name()) {
            var nameVisitor = new NameVisitor.NameVisitor();
            name = nameVisitor.visitName(ctx.name());
        }

        if (ctx.class_modification()) {
            var class_modificationVisitor = new Class_modificationVisitor.Class_modificationVisitor();
            class_modification = class_modificationVisitor.visitClass_modification(ctx.class_modification());
        }

        return new Constraining_clause.Constraining_clause(name, class_modification);
    }
};

Constraining_clauseVisitor.prototype = Object.create(modelicaVisitor.prototype);

exports.visitConstraining_clause = this.visitConstraining_clause;
exports.Constraining_clauseVisitor = Constraining_clauseVisitor;